cmd/swarm/swarm-smoke/main.go

```diff
@@ -110,10 +146,38 @@ func main() {
 
 	sort.Sort(cli.FlagsByName(app.Flags))
 	sort.Sort(cli.CommandsByName(app.Commands))
+	app.Before = func(ctx *cli.Context) error {
+		return nil
+	}
 
 	err := app.Run(os.Args)
 	if err != nil {
 		log.Error(err.Error())
+
+		// wait for metrics reporter to push latest measurements
+		time.Sleep(collectionInterval + 1*time.Second)
+
 		os.Exit(1)
 	}
+
+	// wait for metrics reporter to push latest measurements
+	time.Sleep(collectionInterval + 1*time.Second)
+}
+
+func setupMetrics(ctx *cli.Context) {
+	if gethmetrics.Enabled {
+		var (
+			endpoint = ctx.GlobalString(swarmmetrics.MetricsInfluxDBEndpointFlag.Name)
+			database = ctx.GlobalString(swarmmetrics.MetricsInfluxDBDatabaseFlag.Name)
+			username = ctx.GlobalString(swarmmetrics.MetricsInfluxDBUsernameFlag.Name)
+			password = ctx.GlobalString(swarmmetrics.MetricsInfluxDBPasswordFlag.Name)
+			hosttag  = ctx.GlobalString(swarmmetrics.MetricsInfluxDBHostTagFlag.Name)
+		)
+
+		go influxdb.InfluxDBWithTags(gethmetrics.DefaultRegistry, collectionInterval, endpoint, database, username, password, "swarm-smoke.", map[string]string{
```

Comment:
I am not sure this is the best way to collect measurements for each `swarm-smoke` run.

At the moment we could very well collect measurements only once at the end of the process, rather than emit them every 5 seconds. This has the added value that in the visualization in Grafana, we can link individual measurements to a specific test.

Until we have tests in the `swarm-smoke` process that require periodic sending of metrics, I think we should do it only once - at the end of the job.